rotkehlchen/db/addressbook.py

```diff
@@ -36,7 +36,15 @@ def read_ctx(self, book_type: AddressbookType) -> Generator['DBCursor', None, No
             yield cursor
 
     @contextmanager
-    def write_ctx(self, book_type: AddressbookType) -> Generator['DBCursor', None, None]:
+    def write_ctx(
+            self,
+            book_type: AddressbookType,
+            given_cursor: Optional['DBCursor'] = None,
```

Comment:
This is for situations when some cursor already exists and has to be reused. Since a lock is used for cursors https://github.com/rotki/rotki/blob/ee11e2046badefcbc9578c05e641f410dd03d151/rotkehlchen/db/drivers/gevent.py#L358 we can't have 2 cursors instantiated at the same time.

As for why to pass the cursor here -- it looked logical to me to give to `write_ctx` context manager all the data that is needed to determine which cursor to use.

Reply:
> we can't have 2 cursors instantiated at the same time.

Ohhh, got it.